tests/ert_tests/ert3/data/test_transmitters.py

```diff
@@ -14,10 +15,16 @@
     RecordTransmitter,
     SharedDiskRecordTransmitter,
 )
+import ert
+
+
+@pytest.fixture()
+def storage_path(workspace, ert_storage):
+    yield ert.storage.get_records_url(workspace)
```

Comment:
This dependency on a workspace, will that cause actual files to be written to disk? I think we should be able to run these tests and create a "workspace" for tests without writing files to disk, but I guess this is far beyond the scope of this PR...

Reply:
Because if we could avoid the workspace here we could get the storage path only in the `ert_storage_factory_context`, and not have to pass it to all the contexts.

Reply:
I think the dependency on the worksapce can be avoided, as long as we write an experiment and ensemble to the DB and then get back the ensemble id, but that seems a bit more complicated than just using the workspace setup mechanics

Reply:
Agreed. I think we might need to revisit this behavior in the future to improve it, but some maturation of how we would like it to work is probably needed first.

Reply:
Not really sure if I should try to remove the dependency on the workspace in this PR or not.